contrib/build-push-cephonly-base-images.sh

```diff
@@ -0,0 +1,157 @@
+# corresponding image and tag on the registry, the script will build the image and push it to the
+# registry. The script will do nothing if images are at parity with packages.
+#
+# This script can therefore be run on a cron regularly, and very little manual maintenance should be
+# necessary unless Ceph RPM releases or images published change. The only maintenance needed is to
+# add new Ceph releases to the `FLAVORS_TO_BUILD` variable below.
+
+
+#---------------------------------------------------------------------------------------------------
+# YOU CAN TUNE A SCRIPT, BUT YOU CAN'T TUNE A FISH
+#---------------------------------------------------------------------------------------------------
+
+# This is mostly an internal representation of `FLAVORS_TO_BUILD`. This script builds x86 and arm
+# images for centos, so the middle item doesn't really matter except to make this easier to grok at
+# first glance.
+#   Usage examples:
+# luminous,centos,7 = build luminous containers on centos 7
+# mimic,centos,7 = build mimic containers on centos 7
+# nautilus,centos,8 = build nautilus containers on centos 8
+FLAVORS_TO_BUILD="luminous,centos,7 mimic,centos,7"
+
+# Use the ceph library by default
+PUSH_LIBRARY="ceph"
+
+# Instead of 'daemon-base', call these images 'ceph' so users don't get these confused with the
+# daemon-base and daemon images
+PUSH_IMAGE_NAME="ceph"
+
+# Push to the docker hub by default
+PUSH_REGISTRY_URL="https://registry.hub.docker.com/v2/repositories/${PUSH_LIBRARY}/${PUSH_IMAGE_NAME}"
+
+
+#---------------------------------------------------------------------------------------------------
+# FUNCTIONS
+#---------------------------------------------------------------------------------------------------
+
+# Return a list of the ceph version strings available on the server as:
+#   "12.0.0 12.0.1 12.0.2 12.1.0 ..."
+function ceph_versions_on_server() {
+  ceph_codename="${1}" ; distro_release="${2}" ; ceph_arch="${3}"
+  download_url="http://download.ceph.com/rpm-${ceph_codename}/el${distro_release}/${ceph_arch}/"
+  # Curl gives a listing of the URL like below:
+  # <a href="ceph-12.2.7-0.el7.x86_64.rpm">ceph-12.2.7-0.el7.x86_64.rpm</a>  17-Jul-2018 14:11  3024
+  # The ceph base package can be id'ed uniquely by the text ">ceph-" followed by a version string
+  pkg_regex=">ceph-[0-9.]+"
+  # pkg_list is returned in the form ">ceph-12.0.0 >ceph-12.0.1 >ceph-12.0.2 ..."
+  pkg_list="$(curl --silent "${download_url}" | grep --extended-regexp --only-matching "${pkg_regex}")"
+  version_list=""  # version strings only
+  for pkg in $pkg_list; do
+    version_list="${version_list} ${pkg#>ceph-}"  # strip '>ceph-' text from beginning of pkg names
+  done
+  echo "${version_list}"
+}
+
+# Return 0 if the tag exists at the registry URL; nonzero otherwise
+function tag_exists_on_registry () {
+  tag="${1}"
+  curl --silent --fail --list-only --show-error --location \
+    "${PUSH_REGISTRY_URL}/tags/${tag}" &> /dev/null
```

Comment:
You need to iterate through the pages of the docker hub, this will only report the last 10 tags, I don't think this is enough.

Reply:
This checks for the existence of a single tag with curl. No iteration necessary.

Reply:
Right, read too fast :).